internal/retry/retry.go

```diff
@@ -104,7 +119,7 @@ func (r *OutboundMiddleware) Call(ctx context.Context, request *transport.Reques
 			return resp, err
 		}
 
-		// TODO add backoff semantics
+		time.Sleep(r.getBackoff(i))
```

Comment:
Definitely agree about trying to avoid sleeping in tests. Peter's approach is the cleanest; less clean is to have a `var _sleep = time.Sleep` global and swap it out in tests.